### PR TITLE
star rod dx prep

### PR DESCRIPTION
**Pull request by HailSanta**

### CODE REVIEW

**Review by bates64 — COMMENTED** [review]
The vast majority of this PR should be able to be backported upstream. I'm tempted to block it on that front because we **cannot** afford to diverge too much.

Changes look good tho.

**Comment by bates64** [comment]
That is, please cherry-pick most of these changes upstream, get then merged there, and then we can merge this

File: include/functions.h [hunk 1]
```diff
@@ -270,6 +270,24 @@ s32 test_ray_zones(f32 startX, f32 startY, f32 startZ, f32 dirX, f32 dirY, f32 d
                    f32* hitDepth, f32* nx, f32* ny, f32* nz);
 s32 test_ray_colliders(s32 ignoreFlags, f32 startX, f32 startY, f32 startZ, f32 dirX, f32 dirY, f32 dirZ, f32* hitX,
                        f32* hitY, f32* hitZ, f32* hitDepth, f32* hitNx, f32* hitNy, f32* hitNz);
+
+/// Test a general ray from a given starting position and direction against all entities.
+/// If one is hit, returns the position and normal of the hit and the length along the ray on the output params.
+/// All output params are only valid when a value of `NO_COLLIDER` is returned.
```
Comment (bates64) [inline include/functions.h:276]: I may be misunderstanding but I'd expect the inverse to be true? That is I'd expect the output params are only valid when `NO_COLLIDER` is NOT returned

Comment (HailSanta) [reply include/functions.h:276]: Oops forgot a word there

**Review by bates64 — APPROVED** [review]

**Pull request merged**